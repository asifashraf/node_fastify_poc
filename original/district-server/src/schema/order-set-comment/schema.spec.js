const { fetchGraphQL } = require('../../lib/test-util');
const {
  sampleOrderSetId,
  orderSetCommentDetails,
} = require('../../lib/test-fragments');

test('order set can resolve comments', async () => {
  const query = `{
    orderSet(id: "${sampleOrderSetId}") {
        comments { ${orderSetCommentDetails} }
    }
  }`;
  // openInGraphiql(query);
  const result = await fetchGraphQL(query);
  expect(result).toMatchSnapshot();
});

test('can add a comment for an order set', async () => {
  const mutation = `
      mutation orderSetCommentAdd($orderSetComment: OrderSetCommentInput!)  {
      orderSetCommentAdd(orderSetComment: $orderSetComment) {
        error
        orderSetComment {
          userName
          userName
          comment
        }
      }
    }
  `;

  const variables = {
    orderSetComment: {
      orderSetID: sampleOrderSetId,
      type: 'REQUEST',
      comment: 'This is a Test Comment',
    },
    __user: {
      id: 'testUserId',
      name: 'Tester McTesty',
      email: 'john@sky.net',
    },
  };

  const result = await fetchGraphQL(mutation, variables);
  expect(result).toMatchSnapshot();
});
