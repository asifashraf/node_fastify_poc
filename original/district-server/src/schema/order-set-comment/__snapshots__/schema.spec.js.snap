// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can add a comment for an order set 1`] = `
Object {
  "data": Object {
    "orderSetCommentAdd": Object {
      "error": null,
      "orderSetComment": Object {
        "comment": "This is a Test Comment",
        "userName": "Tester McTesty",
      },
    },
  },
}
`;

exports[`order set can resolve comments 1`] = `
Object {
  "data": Object {
    "orderSet": Object {
      "comments": Array [
        Object {
          "comment": "Nisi perferendis veniam nulla eius iusto et voluptatum dolorem. Dolores quo consequuntur fugit ut laborum veniam modi. Ipsa repellendus veniam excepturi exercitationem. Dolores ducimus quisquam maxime sed atque. Nulla quisquam atque ut asperiores eius nisi ut eaque.",
          "createdAt": "2018-01-05T11:00:00Z",
          "id": "1ccd51d9-8867-437b-a3ee-c8f3e76fd5f3",
          "type": "COMPLAIN",
          "userEmail": "Mackenzie.Rogahn@yahoo.com",
          "userId": "b54595da-14de-4b14-87dd-3972e8535ab8",
          "userName": "Mrs. Tony Lang",
        },
      ],
    },
  },
}
`;
